clients/rust/src/lib.rs

```diff
@@ -68,18 +68,18 @@ impl SysadminClient {
     /// "address" will become a SocketAddr using the trait
     /// `[ToSocketAddrs]`: https://doc.rust-lang.org/std/net/trait.ToSocketAddrs.html
     pub fn connect<A: ToSocketAddrs>(&mut self, address: A) -> Result<()> {
```

Comment:
So, I think everything in this file should be changed to use the internal `SysadminResult` type directly and not through this alias. The reason for defining the alias isn't so you can use it internally, it's for clients to be able to import your result type for use in client code, similar to how the Rust standard library is structured (e.g. `std::fmt::Result`). IMO, the way it is here is confusing due to shadowing `std::result::Result` which is automagically imported in every file via the [prelude](https://doc.rust-lang.org/std/prelude/).